frontend/ui/widget/inputtext.lua

```diff
@@ -141,8 +143,90 @@ if Device:isTouchDevice() or Device:hasDPad() then
                 self.text_widget:moveCursorToXY(x, y, true) -- restrict_to_view=true
                 self.charpos, self.top_line_num = self.text_widget:getCharPos()
```

Comment:
I wonder if we should position the cursor at the place we hold - or if we'd better not let it be where it was, so we can precisely more it with the cursor keys.
When I long-press, my finger might move a little and most often, the past is a few char off.
So, a long-press anywhere might be safer (but may be a little bit unintuitive?)
(Only when not doing start/end selection.)

Reply:
I agree that positioning with hold is not so precious.
But it would be a little bit unusual: "set the cursor to the desired position and then hold anywhere in the screen".
Besides, it would make impossible to call the clipboard dialog with one touch (hold), we would need to position the cursor first (with arrows or with a tap), and then hold the screen.
I'll play with the usability of this dialog a little bit to feel the best way.

Reply:
Wow, it seems to be a great idea: **hold does not move the cursor**.
It's very convenient to put the cursor with a tap or arrows, be sure that it is in the desired position, and then hold anywhere to invoke the dialog or to mark the start/end of the selection.

Reply:
Was going to say: then, we just drop "hold" and do nothing on long-press, and add another button with copy/cut/paste symbols.
But then, it would be available only on big InputDialog with nav bars, not on simple ones...

Reply:
The ideal would be little draggers at the end if you ever feel bored. :-P

https://www.cultofmac.com/ezoimgfmt/cdn.cultofmac.com/wp-content/uploads/2014/02/cutandpaste1.jpg?ezimgfmt=rs:728x689/rscb13/ng:webp/ngcb13

Reply:
(Please ignore this ideal and leave TextBoxWidget alone :)
(Or it would have to be some overlay stuff drawn from somewhere else.)